### PR TITLE
UI-257 Full page stencil Signed-Off-By: Tara Black <tblaack@chef.io>

### PR DESCRIPTION
**Pull request by tarablack01**
### :nut_and_bolt: Description
We need a Stencil component for the full-page design. It is already being used in Settings > Policies > > Add Members and will soon need to be used for ingest rules. Please work with @tylercloke who did the Add Members implementation and @susanev who created the designs.

### :+1: Definition of Done
Stencil component for a full-page
Add members implementation of stencil page component

### :chains: Related Resources
https://github.com/chef/automate/issues/257

### CODE REVIEW

File: components/chef-ui-library/src/templates/chef-page/chef-page.tsx [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+                secondary
+                onClick={this.handleSecondaryConfirm.bind(this)}>
+                {this.secondaryBtnText}
+              </chef-button>
+            : ''
+          }
+
+          <div id="right-buttons">
+
+            {(this.disableConfirm || this.confirmLoading)
```
Comment (msorens) [inline components/chef-ui-library/src/templates/chef-page/chef-page.tsx]: Not clear what purpose disableConfirm does here... seems to be related to my other comment on "Adding Member...".

File: components/chef-ui-library/src/templates/chef-page/chef-page.tsx [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+ *  <chef-table>
+ *    <chef-thead>
+ *       <chef-tr>
+ *         <chef-th class="checkbox-row"></chef-th>
+ *         <chef-th>ID</chef-th>
+ *         <chef-th>Type</chef-th>
+ *       </chef-tr>
+ *    </chef-thead>
+ *    <chef-tbody>
+ *      <chef-tr >
```
Comment (msorens) [inline components/chef-ui-library/src/templates/chef-page/chef-page.tsx]: Remove extra space

File: components/automate-ui/src/app/pages/policy/add-members/policy-add-members.component.html [hunk 3]
```diff
@@ -1,108 +1,85 @@
...
+          </div>
+        </div>
+        <div id="inner-container">
+          <chef-table>
+            <chef-thead>
+              <chef-tr>
+                <chef-th class="checkbox-row"></chef-th>
+                <chef-th>
+                  ID
+                </chef-th>
```
Comment (msorens) [inline components/automate-ui/src/app/pages/policy/add-members/policy-add-members.component.html]: Suggest put element on one line, i.e.: `<chef-th>ID</chef-th>`
Similarly for L57-59.

**Comment by msorens** [comment]
I reviewed from code-reading only; still need someone to try out the Add Policy Members functionality but my system is not in a state where I could do that right now.

**Comment by tarablack01** [comment]
@tylercloke You will need to do a build in the chef-ui-library and a npm install in the automate-ui to get the latest changes for the stencil components.

**Comment by msorens** [comment]
Actually, @tarablack01 and @tylercloke -- getting the latest stencil changes can be done with just one command, in the automate-ui directory: `make update-ui-lib`. (That does the two things that @tarablack01 mentioned above.)

**Pull request merged**